src/cbc_sdk/enterprise_edr/base.py

```diff
@@ -237,6 +246,103 @@ def children(self):
         return [Process(self._cb, initial_data=child) for child in self.nodes["children"]]
 
 
+class ProcessFacet(UnrefreshableModel):
+    """Represents the results of an AsyncFacetQuery.
+
+    ProcessFacet objects contain both Terms and Ranges. Each of those contain facet
+    fields and values.
+
+    Access all of the Terms facet data with ProcessFacet.terms_.facets or see just
+    the field names with ProcessFacet.terms_.fields.
+
+    Access all of the Ranges facet data with ProcessFacet.ranges_.facets or see just
+    the field names with ProcessFacet.ranges_.fields.
+    """
+    primary_key = "job_id"
+    swagger_meta_file = "enterprise_edr/models/process_facets.yaml"
+    urlobject = "/api/investigate/v2/orgs/{}/processes/facet_jobs"
+
+    class Terms(UnrefreshableModel):
+        """Represents the facet fields and values associated with a Process Facet query."""
+        def __init__(self, cb, initial_data):
+            """Initialize a ProcessFacet Terms object with initial_data."""
+            super(ProcessFacet.Terms, self).__init__(
+                cb,
+                model_unique_id=None,
+                initial_data=initial_data,
+                force_init=False,
+                full_doc=True,
+            )
+            self._facets = {}
+            for facet_term_data in initial_data:
```

Comment:
Is there a possibility that `initial_data` is None due to an uncaught error?

Reply:
I don't think so. It could be an empty list `[]`, but not None without a `ServerError` being raised. In `connection.BaseAPI`, the function `get_object` performs a GET on the job results URL with optional query params. If the response from CB isn't code 200, or it's unable to parse the response as JSON, a `ServerError` gets raised.

If the API returned 204, then initial_data could be None. I think a 204 would only come back on a DELETE, otherwise if there's no matching data to return, the API gives back "terms" and "ranges" as empty JSON objects. In that case, we'd have `initial_data == []`

Reply:
Thanks for the explanation